chore: removed radix-ui components

AkTewari123 opened the pull request:
I have removed most of the radix-ui components from the code, except for one called "radix-ui/slot" in common/Button. I didn't remove this since I am not that author of that file, so I don't know what they used it for. 

I will now work on a leaderboard and "friends" feature on the website.

Hemit99123 commented:
Error from vercel, please fix:
```bash 
[21:43:03.842] Running build in Washington, D.C., USA (East) – iad1
[21:43:03.843] Build machine configuration: 2 cores, 8 GB
[21:43:03.860] Cloning github.com/Hemit99123/dailysat (Branch: main, Commit: b6c4cdc)
[21:43:04.913] Cloning completed: 1.052s
[21:43:08.503] Restored build cache from previous deployment (4PP8z1GAzZevVTX8kLyLuR2MA3Hz)
[21:43:09.919] Running "vercel build"
[21:43:10.317] Vercel CLI 41.7.3
[21:43:10.676] Installing dependencies...
[21:43:12.102] npm error code ERESOLVE
[21:43:12.102] npm error ERESOLVE could not resolve
[21:43:12.103] npm error
[21:43:12.103] npm error While resolving: @types/react-dom@19.1.4
[21:43:12.103] npm error Found: @types/react@18.3.18
[21:43:12.103] npm error node_modules/@types/react
[21:43:12.103] npm error   dev @types/react@"^18" from the root project
[21:43:12.103] npm error   peerOptional @types/react@"*" from @radix-ui/react-alert-dialog@1.1.11
[21:43:12.103] npm error   node_modules/@radix-ui/react-alert-dialog
[21:43:12.104] npm error   34 more (@radix-ui/react-primitive, ...)
[21:43:12.104] npm error
[21:43:12.104] npm error Could not resolve dependency:
[21:43:12.104] npm error peer @types/react@"^19.0.0" from @types/react-dom@19.1.4
[21:43:12.104] npm error node_modules/@types/react-dom
[21:43:12.105] npm error   dev @types/react-dom@"^19.1.4" from the root project
[21:43:12.105] npm error   peerOptional @types/react-dom@"*" from @radix-ui/react-alert-dialog@1.1.11
[21:43:12.105] npm error   node_modules/@radix-ui/react-alert-dialog
[21:43:12.105] npm error   14 more (@radix-ui/react-primitive, @radix-ui/react-dialog, ...)
[21:43:12.105] npm error
[21:43:12.105] npm error Conflicting peer dependency: @types/react@19.1.4
[21:43:12.105] npm error node_modules/@types/react
[21:43:12.105] npm error   peer @types/react@"^19.0.0" from @types/react-dom@19.1.4
[21:43:12.106] npm error   node_modules/@types/react-dom
[21:43:12.106] npm error     dev @types/react-dom@"^19.1.4" from the root project
[21:43:12.106] npm error     peerOptional @types/react-dom@"*" from @radix-ui/react-alert-dialog@1.1.11
[21:43:12.106] npm error     node_modules/@radix-ui/react-alert-dialog
[21:43:12.106] npm error     14 more (@radix-ui/react-primitive, @radix-ui/react-dialog, ...)
[21:43:12.106] npm error
[21:43:12.106] npm error Fix the upstream dependency conflict, or retry
[21:43:12.106] npm error this command with --force or --legacy-peer-deps
[21:43:12.107] npm error to accept an incorrect (and potentially broken) dependency resolution.
[21:43:12.107] npm error
[21:43:12.107] npm error
[21:43:12.107] npm error For a full report see:
[21:43:12.107] npm error /vercel/.npm/_logs/2025-05-14T01_43_10_888Z-eresolve-report.txt
[21:43:12.107] npm error A complete log of this run can be found in: /vercel/.npm/_logs/2025-05-14T01_43_10_888Z-debug-0.log
[21:43:12.145] Error: Command "npm install" exited with 1
[21:43:12.722] 
[21:43:15.566] Exiting build container

```

Hemit99123 commented:
No worries, keep the radix for the components that you have not authored.

Hemit99123 commented on src/app/ai/page.tsx:
```diff
@@ -58,52 +59,54 @@ const AI = () => {
         } catch (err) {
-          console.error("Parsing error:", err)
-          setStudyPlan(plan)
+          console.error("Parsing error:", err);
```
why did we add a console.error()?

Hemit99123 replied:
shouldn't the console.error() be removed?

Hemit99123 replied:
i get your concern, but use a toast so that error variable is used

Hemit99123 commented on src/app/ai/page.tsx:
```diff
@@ -58,52 +59,54 @@ const AI = () => {
-      console.error("Error generating study plan:", error)
-      setStudyPlan({ error: "Failed to generate plan", isError: true })
-      setStep(2)
+      console.error("Error generating study plan:", error);
```
again, pls remove it

Hemit99123 commented on src/app/checkout/page.tsx:
```diff
@@ -146,13 +144,14 @@ const Checkout = () => {
   // Render checkout interface
   return (
     <>
+      <ToastContainer></ToastContainer>
```
shouldnt this be in the layout already?

Hemit99123 commented on src/app/shop/page.tsx:
```diff
@@ -192,6 +186,8 @@ export default function Shop() {
   }, [setUser]);
   return (
     <>
+      <ToastContainer></ToastContainer>
```
same thing here, shouldn't this already be in layout? not fully sure, just double checking

Hemit99123 commented on src/app/shop/page.tsx:
```diff
@@ -430,10 +413,10 @@ export default function Shop() {
-                      title: "No items selected",
-                      description: "Please select at least one item to buy",
-                      className: "bg-[#4D68C3] border-none text-white  ",
+                    // toast("You don't have any items");
```
dont comment out, just remove it

Hemit99123 reviewed (changes requested):
Please review my comments **carefully** DM me or comment on PR if you have any questions.

Hemit99123 commented on package.json:
```diff
@@ -42,17 +42,17 @@
     "nodemailer": "^6.9.16",
     "npm": "^11.2.0",
-    "react": "^18",
+    "react": "^19.1.0",
```
this is causing vercel errors. look into it further!

Hemit99123 commented on src/app/ai/page.tsx:
```diff
@@ -1,55 +1,57 @@
+import { Button } from "@/components/common/Button";
+import Error from "next/error";
 
 interface Activity {
```
add this into type folder and import it (to keep consistency)

Hemit99123 commented on src/app/checkout/page.tsx:
```diff
@@ -1,11 +1,12 @@
 "use client";
 
 // Import necessary dependencies and components
```
no need for this comment, pretty obv

Hemit99123 commented on src/components/ui/alert-dialog.tsx:
```diff
@@ -1,135 +1,196 @@
+  );
+};
+
+// Header section
```
no need for this comment, again pretty obv

Hemit99123 commented on src/components/ui/alert-dialog.tsx:
```diff
@@ -1,135 +1,196 @@
+));
+AlertDialogFooter.displayName = "AlertDialogFooter";
 
+// Title component
```
remove this comment and explain more better what is happening here (sort of unclear imo)

Hemit99123 reviewed (changes requested):
Good work. Please review comments and implement necessary changes. There was also a Vercel error which is pasted below:

```bash 
[12:08:04.543] Running build in Washington, D.C., USA (East) – iad1
[12:08:04.543] Build machine configuration: 2 cores, 8 GB
[12:08:04.560] Cloning github.com/Hemit99123/dailysat (Branch: main, Commit: 8050004)
[12:08:05.447] Cloning completed: 886.000ms
[12:08:09.584] Restored build cache from previous deployment (4PP8z1GAzZevVTX8kLyLuR2MA3Hz)
[12:08:10.992] Running "vercel build"
[12:08:11.410] Vercel CLI 41.7.3
[12:08:11.982] Installing dependencies...
[12:08:13.547] npm error code ERESOLVE
[12:08:13.549] npm error ERESOLVE could not resolve
[12:08:13.549] npm error
[12:08:13.550] npm error While resolving: react-katex@3.0.1
[12:08:13.550] npm error Found: react@19.1.0
[12:08:13.550] npm error node_modules/react
[12:08:13.550] npm error   react@"^19.1.0" from the root project
[12:08:13.551] npm error   peer react@"^18 || ^19 || ^19.0.0-rc" from @ai-sdk/react@1.2.8
[12:08:13.551] npm error   node_modules/@ai-sdk/react
[12:08:13.551] npm error     @ai-sdk/react@"1.2.8" from ai@4.3.5
[12:08:13.551] npm error     node_modules/ai
[12:08:13.551] npm error       ai@"^4.3.5" from the root project
[12:08:13.552] npm error   79 more (@cyriacbr/react-split-text, @floating-ui/react-dom, ...)
[12:08:13.552] npm error
[12:08:13.552] npm error Could not resolve dependency:
[12:08:13.552] npm error peer react@">=15.3.2 <=18" from react-katex@3.0.1
[12:08:13.552] npm error node_modules/react-katex
[12:08:13.552] npm error   react-katex@"^3.0.1" from the root project
[12:08:13.553] npm error
[12:08:13.553] npm error Conflicting peer dependency: react@18.3.1
[12:08:13.553] npm error node_modules/react
[12:08:13.553] npm error   peer react@">=15.3.2 <=18" from react-katex@3.0.1
[12:08:13.553] npm error   node_modules/react-katex
[12:08:13.553] npm error     react-katex@"^3.0.1" from the root project
[12:08:13.553] npm error
[12:08:13.554] npm error Fix the upstream dependency conflict, or retry
[12:08:13.554] npm error this command with --force or --legacy-peer-deps
[12:08:13.554] npm error to accept an incorrect (and potentially broken) dependency resolution.
[12:08:13.554] npm error
[12:08:13.554] npm error
[12:08:13.555] npm error For a full report see:
[12:08:13.555] npm error /vercel/.npm/_logs/2025-05-14T16_08_12_225Z-eresolve-report.txt
[12:08:13.557] npm error A complete log of this run can be found in: /vercel/.npm/_logs/2025-05-14T16_08_12_225Z-debug-0.log
[12:08:13.600] Error: Command "npm install" exited with 1
[12:08:14.132] 
[12:08:17.125] Exiting build container
```

AkTewari123 commented:
I have fixed the vercel deployment errors.
I also implemented the other changes
If the image below doesn't load, it's a picture of my vercel fork properly displaying my changes
<img width="1408" alt="Screenshot 2025-05-14 at 7 55 17 PM" src="https://github.com/user-attachments/assets/59c6ec9b-0ecc-4882-8764-486e8cd7fee3" />

Hemit99123 commented on src/app/ai/page.tsx:
```diff
@@ -58,52 +35,58 @@ const AI = () => {
-            setStudyPlan(plan)
+            setStudyPlan(plan);
           }
         } catch (err) {
```
on second thought, instead of err use error variable name

Hemit99123 reviewed (changes requested):
Everything looks good except the console.error in the Study AI route. Look at my new comments and Discord msgs as well.

Hemit99123 commented:
Solve git conflicts

Hemit99123 reviewed (approved):
LGTM 👍
